services/elections.request.service.ts

```diff
@@ -0,0 +1,340 @@
+   */
+  createPosition(data: any): Observable<Position> {
+    const positionsObservable = super.post(`${this.baseURL}/position`, data);
+    return positionsObservable;
+  }
+
+  /**
+   * Update positions
+   *
+   * https://docs.aswwu.com/?url=https://raw.githubusercontent.com/ASWWU-Web/python_server/develop/docs/elections.yml#/position/put_position__position_id_
+   * @param data
+   * @param queryParams
+   * @return JSON object containing the position info that was updated
+   */
+  updatePosition(data: any, queryParams: any): Observable<Position> {
+    const positionObservable = super.put(`${this.baseURL}/position/` + queryParams, data);
+    return positionObservable;
+  }
+
+  ///////////////////
+  // Candidates
+  //////////////////
+  /**
+   * List candidates
+   *
+   * https://docs.aswwu.com/?url=https://raw.githubusercontent.com/ASWWU-Web/python_server/develop/docs/elections.yml#/candidate/get_election__election_id__candidate
+   * @param queryParams
+   * @return an observable of all elections or elections specified by urlParams
+   */
+  listCandidates( queryParams: any, jsonData?: any): Observable<Candidate[]> {
+    const candidateObservable = super.get(`${this.baseURL}/election/${queryParams}/candidate`, jsonData).pipe(
+      map((data: {candidates: Candidate[]}) => data.candidates)
+    );
+    return candidateObservable;
+  }
+
+  /**
+   * Read candidates
+   *
+   * https://docs.aswwu.com/?url=https://raw.githubusercontent.com/ASWWU-Web/python_server/develop/docs/elections.yml#/candidate/get_election__election_id__candidate__candidate_id_
+   * @param queryParams
+   * @param candidateId
+   * @return a single candidate JSON object
+   */
+  readCandidate(queryParams: any, candidateId: any): Observable<Candidate> {
+    const candidateObservable = super.get(`${this.baseURL}/election/${queryParams}/candidate/${candidateId}`).pipe(
+      map((data: {candidates: Candidate}) => data.candidates)
+    );
+    return candidateObservable;
+  }
+
+  /**
+   * Create candidate
+   *
+   * https://docs.aswwu.com/?url=https://raw.githubusercontent.com/ASWWU-Web/python_server/develop/docs/elections.yml#/candidate/post_election__election_id__candidate
+   * @param queryParams
+   * @return JSON object containing the candidate info created
+   */
+  createCandidate( queryParams: any, data: any): Observable<Candidate> {
+    const candidatesObservable = super.post(`${this.baseURL}/election/${queryParams}/candidate`, data);
+    return candidatesObservable;
+  }
+
+  /**
+   * Update Candidate
+   *
+   * https://docs.aswwu.com/?url=https://raw.githubusercontent.com/ASWWU-Web/python_server/develop/docs/elections.yml#/candidate/put_election__election_id__candidate__candidate_id_
+   * @param data
+   * @param queryParams
+   * @param candidateID
+   * @return JSON object containing the candidate info that was updated
+   */
+  updateCandidate(data: any, queryParams: any, candidateID: any): Observable<Candidate> {
```

Comment:
In a put request we send the whole object to be updated, so we can use the Candidate type for data. this is another case where queryParams is actually electionId
```suggestion
  updateCandidate(data: Candidate, electionID: string, candidateID: any): Observable<Candidate> {
```